docs/architecture/adr-056-proving-amnesia-attack.md

```diff
@@ -0,0 +1,106 @@
+
+The evidence should contain the precommit votes for the intersection of validators that voted for both rounds. The votes should be all valid and the height and time that the infringement was made should be within:
+
+`Bonding period - Amnesia trial period - Gossip safety margin`
+
+With reference to the honest nodes, C1 and C2, in the schematic, C2 will not PRECOMMIT an earlier round, but it is likely, if a node in C1 were to recieve +2/3 PREVOTE's or PRECOMMIT's for a higher round, that it would remove the lock and PREVOTE and PRECOMMIT for the later round. Therefore, unfortunately it is not a case of simply punishing all nodes that have double voted in the `PotentialAmnesiaEvidence`.
+
+Instead we use the Proof of Lock Change (PoLC) referred to in the [consensus spec](https://github.com/tendermint/spec/blob/master/spec/consensus/consensus.md#terms). When an honest node votes again for a later round, in very rare cases, it will generate the PoLC and store it in the evidence reactor for a time equal to the `Bonding Period`
+
+```
+type ProofOfLockChange struct {
+	Votes []*types.Vote
+}
+```
+
+This can be either evidence of +2/3 PREVOTES or PRECOMMITS (either warrants the honest node the right to vote) and is valid, among other checks, so long as the PRECOMMIT vote of the node in V2 came after all the votes in the `ProofOfLockChange` i.e. it received +2/3  votes and then voted itself (F is unable to prove this).
+
+In the event that an honest node receives `PotentialAmnesiaEvidence` it will first `Verify()` it and then will check if it is among the suspected nodes in the evidence. If so, it will retrieve the `ProofOfLockChange` and combine it with `PotentialAmensiaEvidence` to form `AmensiaEvidence`:
+
+```
+type AmnesiaEvidence struct {
+	Evidence *types.PotentialAmnesiaEvidence
+	Proofs	 []*types.ProofOfLockChange
+}
+```
+
+If the node is not required to submit any proof than it will simply broadcast the `PotentialAmnesiaEvidence` .
+
+In either case, in a network where each node has n peers, if n is high enough to easily overcome network problems such as dropout than it should be sufficient that peers only broadcast once per `Amnesia trial period` + `Gossip safety margin` as long as `AmnesiaEvidence` has not yet been committed on the chain.
+
+If a node does not have sufficient peers then it may proportionally send more messages at an appropriate interval.
+
+When a node has successfully validated `PotentialAmnesiaEvidence` it timestamps it and refuses to receive the same form of `PotentialAmnesiaEvidence`. If a node receives `AmnesiaEvidence` it checks it against any current `AmnesiaEvidence` it might have and if so merges the two by adding the proofs, if it doesn't have it yet it run's `Verify()` and stores it.
+
+There can only be one `AmnesiaEvidence` and one `PotentialAmneisaEvidence` stored for each attack.
```

Comment:
one per single height?

Reply:
Yes, I believe it's not possible to have multiple amnesia attacks at a single height. But yes I can change "for each attack" to "for each height"

Reply:
> Yes, I believe it's not possible to have multiple amnesia attacks at a single height.

very much possible. 1/3+ malicious nodes can create unlimited number of forks

Reply:
I still don't understand this unlimited forks thing. My understanding is a fork is when two honest nodes have different states. If we have n - f honest nodes then that set's the upper bound of forks.

Reply:
I meant light forks I guess. but one can commit an amnesia attack w/ light fork. there will be no actual fork though.